activejob/lib/active_job/queue_adapters.rb

```diff
@@ -48,5 +48,25 @@ module QueueAdapters
     autoload :SneakersAdapter
     autoload :SuckerPunchAdapter
     autoload :TestAdapter
+
+    ADAPTER = 'Adapter'.freeze
+
+    class << self
+      def instance(name, *adapter_args, &block)
+        cache.compute_if_absent(([name, block] << adapter_args).hash) do
+          lookup(name).new(*adapter_args, &block)
+        end
+      end
```

Comment:
Caching is fine, but worth considering starting without it. The caller is asking for an instance of the named queue adapter — any instance.

Maybe we should dial this back to a simpler "here's an adapter name, give me its class" interaction to start. Then `#interpret_adapter` calls `ActiveJob::QueueAdapters.lookup(name_or_adapter_or_class)` (or `.fetch` or `#[]` or …) and calls `.new` itself.

We can tackle resource building, pooling, etc mechanics later when we have more pressure exerted on them.